src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -340,8 +340,11 @@ public void Close()
 
         void Cleanup()
         {
-            Log.Debug("COPY operation ended", _connector.Id);
-            _connector.CurrentCopyOperation = null;
+            Log.Debug("COPY operation ended", _connector?.Id ?? -1);
+
+            if (_connector != null)
+                _connector.CurrentCopyOperation = null;
+
             _connector = null;
```

Comment:
You can move it into the conditional block above to skip assigning `null` when the value is `null` already.